apps/hubspot/functions/appEventHandler.ts

```
@@ -0,0 +1,69 @@
+import {
+  FunctionEventHandler,
+  FunctionTypeEnum,
+  FunctionEventContext,
+  AppEventRequest,
+} from '@contentful/node-apps-toolkit';
+import {
+  ConnectedFields,
+  EntryConnectedFields,
+  initContentfulManagementClient,
+} from '../src/utils/utils';
+import ConfigEntryService from '../src/utils/ConfigEntryService';
+
+const WAIT_TIMES = [1000, 2000];
+
+export const handler: FunctionEventHandler<FunctionTypeEnum.AppEventHandler> = async (
+  event: AppEventRequest,
+  context: FunctionEventContext
+) => {
+  const cma = initContentfulManagementClient(context);
+
+  const contentfulTopic = event.headers['X-Contentful-Topic'];
+
+  const body = event.body as any;
+  const entryId = body.sys.id;
+
+  const configService = new ConfigEntryService(cma);
+  let connectedFields: ConnectedFields;
+  let entryConnectedFields: EntryConnectedFields;
+  try {
+    connectedFields = await configService.getConnectedFields();
+    entryConnectedFields = await configService.getEntryConnectedFields(entryId);
+  } catch (error) {
+    return;
+  }
```

If this fails we probably want the function to throw the error instead of failing silently. And since this can throw, it may be useful to wrap it in the callAndRetry call (ideally we don't have two separate calls to callAndRetry and we find a way to wrap everything toghether)